plugins/modules/win_dns_record.ps1

```diff
@@ -123,8 +130,20 @@ if ($null -ne $values -and $values.Count -gt 0) {
     foreach ($value in $values) {
         $splat_args = @{ $type = $true; $record_argument_name = $value }
         $module.Result.debug_splat_args = $splat_args
+        $srv_args = @{ ZoneName = $zone
+                       DomainName = $value
+                       Weight = $weight
+                       Priority = $priority
+                       Port = $port
+                       TimeToLive = $ttl
+        }
         try {
-            Add-DnsServerResourceRecord -ZoneName $zone -Name $name -AllowUpdateAny -TimeToLive $ttl @splat_args -WhatIf:$module.CheckMode @extra_args
+            if ($type -ne 'SRV') {
+                Add-DnsServerResourceRecord -ZoneName $zone -Name $name -AllowUpdateAny -TimeToLive $ttl @splat_args -WhatIf:$module.CheckMode @extra_args
+            }
+            else {
+                Add-DnsServerResourceRecord -SRV -Name $name @srv_args @extra_args -WhatIf:$module.CheckMode
+            }
```

Comment:
It's a bit easier to maintain code when the positive condition is being checked instead of the inverse, i.e.

```suggestion
            if ($type -eq 'SRV') {
                Add-DnsServerResourceRecord -SRV -Name $name @srv_args @extra_args -WhatIf:$module.CheckMode
            }
            else {
                Add-DnsServerResourceRecord -ZoneName $zone -Name $name -AllowUpdateAny -TimeToLive $ttl @splat_args -WhatIf:$module.CheckMode @extra_args
            }
```